### PR TITLE
Issue 115 Adding Design components to Tutorial View.

### PR DESCRIPTION
**Pull request by amymariewall**

### CODE REVIEW

File: app/views/tutorials/view.html.erb [hunk 1]
```diff
@@ -1,9 +1,28 @@
-<div class="tutorial_info"> 
-  <h1><%= @tutorial.title %></h1>
-  <%= @tutorial.get_privacy_notice %><br>
-   <%=  @tutorial.get_category_notice %> <br>
-  <%= @tutorial.description %><br><br>
-</div>
+<div class="container">
+
+  <div class="page-header tutorial_info">
+    <h1><%= @tutorial.title %>
```
Comment (sumeetjain) [inline app/views/tutorials/view.html.erb:48]: Remove trailing whitespace

File: app/presenters/tutorial_view_chapter_presenter.rb [hunk 2]
```diff
@@ -13,6 +13,14 @@ def content
        "Chapter #{@count}: #{chapter.title} - (#{chapter.video_time_conversion_for_chapters})"
   end
 
+  def chapter_name
+    "#{chapter.title}"
```
Comment (sumeetjain) [inline app/presenters/tutorial_view_chapter_presenter.rb:17]: Prefer `to_s` over string interpolation.

File: app/presenters/tutorial_view_chapter_presenter.rb [hunk 3]
```diff
@@ -13,6 +13,14 @@ def content
...
+  end
+
+  def chapter_number
+    "#{@count}"
```
Comment (sumeetjain) [inline app/presenters/tutorial_view_chapter_presenter.rb:21]: Prefer `to_s` over string interpolation.

**Comment by amymariewall** [comment]
I'm not sure if I installed Bootstrap correctly. I chose to download the CSS version and not the SASS version, which I think may have been a mistake. I also was not sure if I could keep any of the Base Grid stylesheet files or if I should delete them all when I installed Bootstrap.

File: app/assets/stylesheets/application.scss [hunk 4]
```diff
@@ -7,4 +7,9 @@
 
 // Import additional stylesheets as you make them.
 @import "video-js";
-@import "tutorials";
\ No newline at end of file
+@import "tutorials";
+@import "bootstrap.min";
+@import "bootstrap";
```
Comment (sumeetjain) [inline app/assets/stylesheets/application.scss]: You only need bootstrap.min–or bootstrap–but not both. They're the same. The .min version is just a compressed file.

File: app/assets/stylesheets/application.scss [hunk 5]
```diff
@@ -7,4 +7,9 @@
...
+@import "bootstrap-theme";
+@import "bootstrap-theme.min";
```
Comment (sumeetjain) [inline app/assets/stylesheets/application.scss:12]: Same, only need one of these.

**Comment by sumeetjain** [comment]
> I chose to download the CSS version and not the SASS version, which I think may have been a mistake.

That shouldn't cause problems.

> I also was not sure if I could keep any of the Base Grid stylesheet files or if I should delete them all when I installed Bootstrap.

Get rid of 'em.

**Comment by amymariewall** [comment]
Thanks! One more question - there was a gem to install bootstrap. Is that just another way of doing it or should I have used it?

**Comment by sumeetjain** [comment]
Just another way of doing it. It basically just does what you did manually. I prefer the manual approach.

**Comment by amymariewall** [comment]
Got it, thank you.

File: app/models/progress.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+class Progress < ActiveRecord::Base
```
Comment (sumeetjain) [inline app/models/progress.rb:1]: Has no descriptive comment (IrresponsibleModule)

File: db/migrate/20160413162134_create_progresses.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+class CreateProgresses < ActiveRecord::Migration
```
Comment (sumeetjain) [inline db/migrate/20160413162134_create_progresses.rb:1]: Has no descriptive comment (IrresponsibleModule)

File: db/migrate/20160413162134_create_progresses.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+  def change
+    create_table :progresses do |t|
```
Comment (sumeetjain) [inline db/migrate/20160413162134_create_progresses.rb:3]: Has the variable name 't' (UncommunicativeVariableName)

File: db/migrate/20160413162315_add_user_ref_to_progress.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+class AddUserRefToProgress < ActiveRecord::Migration
```
Comment (sumeetjain) [inline db/migrate/20160413162315_add_user_ref_to_progress.rb:1]: Has no descriptive comment (IrresponsibleModule)

File: db/migrate/20160413162134_create_progresses.rb [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+
```
Comment (sumeetjain) [inline db/migrate/20160413162134_create_progresses.rb:4]: Extra empty line detected at block body beginning.

File: db/migrate/20160413162315_add_user_ref_to_progress.rb [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+  def change
+    add_reference :progresses, :user, index: true, foreign_key: true
+    add_reference :progresses, :step, index: true, foreign_key: true
+
```
Comment (sumeetjain) [inline db/migrate/20160413162315_add_user_ref_to_progress.rb:5]: Extra empty line detected at method body end.

File: db/schema.rb [hunk 12]
```diff
@@ -11,7 +11,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20160325032557) do
+ActiveRecord::Schema.define(version: 20160413162315) do
```
Comment (sumeetjain) [inline db/schema.rb:14]: Separate every 3 digits in the integer portion of a number with underscores(_).

File: db/schema.rb [hunk 13]
```diff
@@ -66,6 +66,16 @@
...
   add_index "delayed_jobs", ["priority", "run_at"], name: "delayed_jobs_priority", using: :btree
 
+  create_table "progresses", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.integer  "user_id"
+    t.integer  "step_id"
+  end
+
+  add_index "progresses", ["step_id"], name: "index_progresses_on_step_id", using: :btree
```
Comment (sumeetjain) [inline db/schema.rb:76]: Line is too long. [89/80]

File: db/schema.rb [hunk 14]
```diff
@@ -66,6 +66,16 @@
...
+  add_index "progresses", ["user_id"], name: "index_progresses_on_user_id", using: :btree
```
Comment (sumeetjain) [inline db/schema.rb:77]: Line is too long. [89/80]

File: app/views/tutorials/view.html.erb [hunk 15]
```diff
@@ -1,48 +1,114 @@
-<div class="tutorial_info"> 
-  <h1><%= @tutorial.title %></h1>
-  <%= @tutorial.get_privacy_notice %><br>
-   <%=  @tutorial.get_category_notice %> <br>
-  <%= @tutorial.description %><br><br>
+
```
Comment (sumeetjain) [inline app/views/tutorials/view.html.erb]: Remove trailing whitespace

File: app/views/tutorials/view.html.erb [hunk 16]
```diff
@@ -1,50 +1,123 @@
...
+          <p class="vjs-no-js">
+            To view this video please enable JavaScript, and consider upgrading to a web browser that
+            <a href="http://videojs.com/html5-video-support/" target="_blank">supports HTML5 video</a>
+          </p>
+
+        </video>
+  </div>
+
+</div>
```
Comment (sumeetjain) [inline app/views/tutorials/view.html.erb:35]: Remove trailing whitespace

File: app/views/tutorials/view.html.erb [hunk 17]
```diff
@@ -1,50 +1,126 @@
...
+          <p class="vjs-no-js">
+            To view this video please enable JavaScript, and consider upgrading to a web browser that
+            <a href="http://videojs.com/html5-video-support/" target="_blank">supports HTML5 video</a>
+          </p>
+
+        </video>
+    </div>
+  </div>
+
+</div>
```
Comment (sumeetjain) [inline app/views/tutorials/view.html.erb:35]: Remove trailing whitespace

File: app/views/tutorials/view.html.erb [hunk 18]
```diff
@@ -1,50 +1,141 @@
...
-  <h1><%= @tutorial.title %></h1>
-  <%= @tutorial.get_privacy_notice %><br>
-   <%=  @tutorial.get_category_notice %> <br>
-  <%= @tutorial.description %><br><br>
-</div>
 
+<!-- Video Lightbox -->
+
+<div id="videoLightBox" class="modal">
+
```
Comment (sumeetjain) [inline app/views/tutorials/view.html.erb:5]: Remove trailing whitespace

**Pull request merged**